test/functional/terminal/tui_spec.lua

```diff
@@ -870,26 +870,55 @@ describe('TUI background color', function()
     feed_data('\027]11;rgb:ffff/ffff/ffff\007')
     screen:expect{any='did OptionSet, yay!'}
   end)
+end)
+
+describe('TUI deferred background color', function()
+  local screen
+
+  setup(function()
+    clear()
+    screen = thelpers.screen_setup(0, '["'..nvim_prog
+      ..'", "-u", "NONE", "-i", "NONE", "--cmd", "set noswapfile"]')
+  end)
+
+  local last_bg = 'dark'
+  local function wait_for_bg(bg)
+    -- Retry until the terminal response is handled.
+    retry(100, nil, function()
+      feed_data(':echo &background\n')
+      screen:expect({
+        timeout=40,
+        grid=string.format([[
+          {1: }                                                 |
+          {4:~                                                 }|
+          {4:~                                                 }|
+          {4:~                                                 }|
+          {5:[No Name]                       0,0-1          All}|
+          %-5s                                             |
+          {3:-- TERMINAL --}                                    |
+        ]], bg)
+      })
+    end)
+    last_bg = bg
+  end
 
   local function assert_bg(colorspace, color, bg)
     it('handles '..color..' as '..bg, function()
```

Comment:
since you're only using one session, just put all of the `assert_bg()` in a single `it()`. then we don't need a separate `describe()` and the `setup()` stuff can go to the start of the single `it()`.

Reply:
Having `it()` in the function allowed for separate descriptions, but I think it is good to get rid of it.